### PR TITLE
Update kubevirt builder image to use go1.13.14

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:

-  Updated the Kubevirt builder image to use go1.13.14.
    
- We now use go modules to get all the dependencies we need.
    It allows us to just go get the specific version that we need instead of having to download and compile it ourselves.
  
-   Set the versions of the dependencies statically.   

- Updated tests to call testing.Init() explicitly in our init() 
functions to use the default flags from the testing package.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Second take of #3420

**Release note**:

```release-note
KubeVirt is now being built with Go1.13.14
```

### CODE REVIEW

**Comment by xpivarc** [comment]
@danielBelenky @davidvossel @rmohr Hopefully we can get this in.

File: WORKSPACE [hunk 1]
```diff
@@ -194,7 +194,7 @@ load(
 go_rules_dependencies()
 
 go_register_toolchains(
-    go_version = "1.12.8",
+    go_version = "1.13.4",
```
Comment (xpivarc) [inline WORKSPACE:197]: This needs to be updated manually?

Comment (rmohr) [reply WORKSPACE:197]: Yes. This tells bazel which golang to use. It does not use the installed golang version. It has its own hermetic packages.

**Comment by xpivarc** [comment]
> Is there a reason to stick with the 1.13 series? 1.14 is out and go1.14.6 is the latest stable release
I would prefer to not skip a major version. Also, it can prolong the update and slow down other PRs(which depends on this one). We found some issues by updating to 1.13 and I think we will find some on 1.14

**Comment by xpivarc** [comment]
/hold 

test_id:3238 seems to fail consistently... I will try to debug it.

**Comment by davidvossel** [comment]
> I would prefer to not skip a major version. Also, it can prolong the update and slow down other PRs(which depends on this one). We found some issues by updating to 1.13 and I think we will find some on 1.14

if it's easier at this point to stabilize on 1.13 that's fine. This step wise approach might make it easier to increment to 1.14 soon. whatever works, I was mostly curious. I definitely don't want us to do anything that might hold up getting this in. this pr is pretty important.

**Comment by xpivarc** [comment]
/hold cancel

The test seems to be flaky and fails on OOMKilled so I increased the memory limit. (I have seen it on multiple PRs)

**Comment by rmohr** [comment]
/lgtm
/approve
/hold

Let's wait for another CI round before we unconditionally want it merged.

**Comment by xpivarc** [comment]
/retest

Seems unrelated and Travis is hitting timeouts?

File: cmd/container-disk-v2alpha/main_test.go [hunk 2]
```diff
@@ -16,6 +17,7 @@ import (
 var containerDiskBinary string
 
 func init() {
+	testing.Init()
```
Comment (EdDev) [inline cmd/container-disk-v2alpha/main_test.go:20]: This is not a recommended fix. The problem is the call to flag.Parse which is wrong.

flag.Parse Is called by the generated test main, all others should just register the flags.

**Comment by rmohr** [comment]
/hold cancel

look good.

/retest

**Comment by EdDev** [comment]
I would appreciate feedback on https://github.com/kubevirt/kubevirt/pull/3945#discussion_r467520465

/hold

File: cmd/container-disk-v2alpha/main_test.go (hunk 2 above)
Comment (xpivarc) [reply cmd/container-disk-v2alpha/main_test.go:20]: First of all, sorry for not replying, had lots of emails from GitHub(with fail tests).

I am not author of this commit but I try my best to answer.
I believe this "fix" was based on [this](https://golang.org/doc/go1.13#testing). It was intended to keep it function as before as this is not the goal of this PR. 

I see this as a specific case when we want to parse the args to be able to modify it. 
I can see that this can be achieved in BeforeEach as follow:
`if !strings.Contains(containerDiskBinary, ../../){
   	containerDiskBinary = filepath.Join("../../", containerDiskBinary)
}`

I fail to see why this is a prefered way to do this. Can you explain what negatives does this brings?

Comment (EdDev) [reply cmd/container-disk-v2alpha/main_test.go:20]: This line will explicitly call the parsing on pre-main which is wrong as it will mask all other flags registrations from other places.
The way flags work is that each library/module/package registers flags they support and in main, once, you call parse in order to process them all. If you call them before, other flags from other sources are not included and even though parse is called again in the test main, it will have no affect (parse is a singleton, secondary calls have no meaning).

Bottom line, there is no need to call this explicitly here, it will be called by the generated test main.

Comment (EdDev) [reply cmd/container-disk-v2alpha/main_test.go:20]: Ref: https://github.com/kubevirt/kubevirt/pull/3786

Comment (xpivarc) [reply cmd/container-disk-v2alpha/main_test.go:20]: I don't see that `Init`calls parsing. Also, I don't see that `flags.Parse()` is singleton (
// Parse parses the command-line flags from os.Args[1:]. Must be called
// after all flags are defined and before flags are accessed by the program.).

I think that we would have failing tests if there was some masking of flags. (This would be the negative impact)
I will align 2 test which left to be aligned with #3786.

Comment (EdDev) [reply cmd/container-disk-v2alpha/main_test.go]: > I don't see that Init calls parsing.

Right, it just registers the test framework flags and is called from main. More reason not to call it here, as it is called from the generated main.

> I don't see that flags.Parse() is singleton

Right, it is the testing.Init who is a singleton and in general the test main will not call `Parse` if it was already called.

The main point is: "Must be called after all flags are defined and before flags are accessed", so you cannot call parse before main, it should be called only at main (after all other inits have been processed).

File: hack/builder/Dockerfile [hunk 3]
```diff
@@ -12,94 +12,63 @@ RUN dnf install -y dnf-plugins-core && \
...
-        findutils \
-        gradle \
-        rsync-daemon \
-        rsync \
-        qemu-img \
-        protobuf-compiler \
-        python3-devel \
-        redhat-rpm-config \
-        jq && \
+    libvirt-devel-${LIBVIRT_VERSION} \
```
Comment (EdDev) [inline hack/builder/Dockerfile:15]: nit: The indentation got lost and it helps readability.

**Comment by EdDev** [comment]
@xpivarc travis frequently fails. Maybe you need to rebase?

**Comment by xpivarc** [comment]
/hold 
Travis fails because ldflags are not passed correctly.

File: hack/common.sh [hunk 4]
```diff
@@ -4,6 +4,8 @@ if [ -f cluster-up/hack/common.sh ]; then
     source cluster-up/hack/common.sh
 fi
 
+export GOFLAGS=-mod=vendor
```
Comment (rmohr) [inline hack/common.sh:7]: Can you just move out all these extra mentions of `GOPROXY=off GOFLAGS=-mod-vendor` to this place and temove them there?

Comment (xpivarc) [reply hack/common.sh:7]: Exporting `GOPROXY=off` will also disable it for bazel... is it intended?

File: hack/version.sh [hunk 5]
```diff
@@ -96,7 +96,7 @@ function kubevirt::version::ldflag() {
     local key=${1}
     local val=${2}
 
-    echo "-X kubevirt.io/kubevirt/vendor/kubevirt.io/client-go/version.${key}=${val}"
+    echo "-X kubevirt.io/client-go/version.${key}=${val}"
```
Comment (danielBelenky) [inline hack/version.sh:99]: This is the wrong place to take it. You should take it from the `vendor/` directory and not use Go modules during the build.

File: pkg/virt-launcher/monitor_test.go [hunk 6]
```diff
@@ -114,6 +112,9 @@ var _ = Describe("VirtLauncher", func() {
 	}
 
 	BeforeEach(func() {
+		if !strings.Contains(fakeQEMUBinary, "../../") {
```
Comment (danielBelenky) [inline pkg/virt-launcher/monitor_test.go:115]: Why this change was needed?

File: tests/utils.go [hunk 7]
```diff
@@ -3606,7 +3606,7 @@ func CreateISCSITargetPOD(containerDiskName cd.ContainerDisk) (iscsiTargetIP str
 	image := fmt.Sprintf("%s/cdi-http-import-server:%s", flags.KubeVirtUtilityRepoPrefix, flags.KubeVirtUtilityVersionTag)
 	resources := k8sv1.ResourceRequirements{}
 	resources.Limits = make(k8sv1.ResourceList)
-	resources.Limits[k8sv1.ResourceMemory] = resource.MustParse("256M")
+	resources.Limits[k8sv1.ResourceMemory] = resource.MustParse("512M")
```
Comment (danielBelenky) [inline tests/utils.go:3609]: Why we need to increase the memory requirements in the tests for Go 1.13?

Comment (xpivarc) [reply tests/utils.go:3609]: https://github.com/kubevirt/kubevirt/pull/3945#issuecomment-670357869

File: pkg/virt-launcher/monitor_test.go (hunk 6 above)
Comment (xpivarc) [reply pkg/virt-launcher/monitor_test.go:115]: https://github.com/kubevirt/kubevirt/pull/3945#discussion_r467520465

File: hack/version.sh (hunk 5 above)
Comment (xpivarc) [reply hack/version.sh:99]: This would not work with 1.13(I am not sure why).
The right format is `importpath.name=string` from [what](https://tip.golang.org/cmd/link/) I know about ldflags. It should not matter that it is vendored.

Comment (danielBelenky) [reply hack/version.sh:99]: It matters because if you're vendoring, you shouldn't even have the package in `kubevirt.io/client-go/version`.

File: pkg/virt-launcher/monitor_test.go (hunk 6 above)
Comment (danielBelenky) [reply pkg/virt-launcher/monitor_test.go:115]: How is this related to the flags we parse? Also, this approach seems wrong to me... if the fakeQEMUBinary doesn't exist exactly at the specified path, we should fail and not try to recover by prefixing the path with `../../`

**Comment by davidvossel** [comment]
how close are we on this PR now? what's left?

**Comment by xpivarc** [comment]
> how close are we on this PR now? what's left?

Waiting for the CI. We are close if nothing shows up.
The migration of goveralls helped a lot!

**Comment by xpivarc** [comment]
/retest

seems checks are not reported even though jobs finished.

File: hack/version.sh (hunk 5 above)
Comment (xpivarc) [reply hack/version.sh:99]: As stated [here](https://golang.org/cmd/go/#hdr-Maintaining_module_requirements)
`If invoked with -mod=vendor, the go command loads packages from the main module's vendor directory instead of downloading modules to and loading packages from the module cache. The go command assumes the vendor directory holds correct copies of dependencies, and it does not compute the set of required module versions from go.mod files. However, the go command does check that vendor/modules.txt (generated by 'go mod vendor') contains metadata consistent with go.mod. `

**Comment by xpivarc** [comment]
/test pull-kubevirt-goveralls

testing theory
